### PR TITLE
CPR-478 Add log events to api

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

**Review by JunaedAhmedMOJ — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt [hunk 1]
```diff
@@ -10,34 +10,45 @@ import uk.gov.justice.digital.hmpps.personrecord.api.model.PersonIdentifierRecor
...
-    else -> throw PersonRecordNotFoundException(identifier)
+  private fun handlePersonRecord(personEntity: PersonEntity?, identifier: Pair<EventKeys, String>): List<PersonIdentifierRecord> = when {
+    personEntity != PersonEntity.empty -> buildListOfLinkedRecords(personEntity!!, identifier)
+    else -> throw PersonRecordNotFoundException(identifier.second)
   }
 
-  private fun buildListOfLinkedRecords(personEntity: PersonEntity): List<PersonIdentifierRecord> {
+  private fun buildListOfLinkedRecords(personEntity: PersonEntity, identifier: Pair<EventKeys, String>): List<PersonIdentifierRecord> {
+    telemetryService.trackEvent(
+      TelemetryEventType.CPR_SEARCH_REQUEST,
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt:46]: What does this give us over a simple appInsights query on `requests`?
I'm not sure this adds any value

Comment (H-Iwanejko) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt:46]: Only value is seeing what UUID is returned as part of the request.

Comment (H-Iwanejko) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt:46]: In the case that we dont want to be raising any excess app insights events. I'll close the PR and we can just query to find the UUID from the record

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt:46]: if it's useful to be able to find out what UUID is returned then fair enough

**Pull request closed without merging**